pkg/kubectl/cmd/util/factory.go

```diff
@@ -554,10 +557,6 @@ func recordCommand(args []string) string {
 	return strings.Join(args, " ")
```

Comment:
this is still mutating os.Args. something like this would be better:
```
	if len(os.Args) == 0 {
		return ""
	}
	base := filepath.Base(os.Args[0])
	args := append([]string{base}, os.Args[1:]...)
	return strings.Join(args, " ")
```